asab/web/auth/service.py

```diff
@@ -0,0 +1,435 @@
+import base64
+import binascii
+import functools
+import inspect
+import json
+import logging
+import typing
+
+import aiohttp
+import aiohttp.web
+import aiohttp.client_exceptions
+
+import asab
+import asab.exceptions
+
+try:
+	import jwcrypto.jwk
+	import jwcrypto.jwt
+	import jwcrypto.jws
+except ModuleNotFoundError:
+	jwcrypto = None
+
+#
+
+L = logging.getLogger(__name__)
+
+#
+
+# Mock user info used in dev mode
+DEV_USERINFO_DEFAULT = {
+	"iss": "auth.test.loc",      # Token issuer
+	"iat": 1682077901,         # Token issued at (timestamp)
+	"exp": 1682092289,         # Token expires at (timestamp)
+	"azp": "my-asab-app",        # Authorized party
+	"aud": "my-asab-app",        # Audience
+	"sub": "abc:xyz:799b53e0",   # Subject (Unique user ID)
+	"preferred_username": "little-capybara",  # Subject's preferred username
+	"email": "capybara1999@example.com",      # Subject's email
+	# Authorized tenants and resources
+	"resources": {
+		# Globally granted resources
+		"*": [
+			"wisdom:access"
+		],
+		# Resources granted within test-tenant
+		"test-tenant": [
+			"wisdom:access",
+			"cake:admire",
+			"cake:smell",
+			"cake:eat",
+		],
+	},
+	# Subject's assigned (not authorized!) tenants
+	"tenants": [
+		"test-tenant",
+		"another-tenant",
+	]
+}
+
+SUPERUSER_RESOURCE = "authz:superuser"
+
+asab.Config.add_defaults({
+	"auth": {
+		# URL location containing the authorization server's public JWK keys
+		"public_keys_url": "",
+
+		# Whether the app is tenant-aware
+		"multitenancy": "yes",
+
+		# In DEV MODE
+		# - no authorization server is needed,
+		# - all incoming requests are "authorized" with custom mocked user info data loaded from a JSON file,
+		"dev_mode": "no",
+		"dev_user_info_path": "",
```

Comment:
@eliska-n (I can't reply directly to your comment)
Yes, the dev mode works by injecting mock userinfo to every request. If you want to access everything, you need to overwrite the default userinfo by providing your own superuser userinfo file. 
You specify `dev_user_info_path=/path/to/userinfo.json` and inside that file just put
```json
{
 "resources": {
  "*": ["authz:superuser"],
  "my-tenant": ["authz:superuser"],
  "another-tenant": ["authz:superuser"],
 }
}
```

Reply:
Now that I think about it, I'll add the superuser resource to the default userinfo. That will be the most common usage anyways.